### PR TITLE
Mocked tests: Better error handling in case port is taken (3000)

### PR DESCRIPTION
**Pull request by catenocrypt**
Fixes issue #1046

### CODE REVIEW

**Comment by kolya182** [comment]
What if we check port Dyson runs on before actually start Dyson ?.
Something like
```
ifeq (false,$(call is_ports_open,3000,8420))
    @echo "  >  Ports 3000 or 8420 are unavailable "
    @exit 1
else
	@exit 1
	@echo "  >  Running $(test) tests"
	@newman run tests/postman/Blockatlas.postman_collection.json --folder $(test) -d tests/postman/$(test)_data.json --env-var "host=$(host)"
    @newman run tests/postman/Blockatlas.postman_collection.json --folder $(test) -d tests/postman/$(test)_data.json --env-var "host=$(host)"
endif
```
define is_ports_open
    ... Check if port not take
endef

**Comment by catenocrypt** [comment]
Is that really needed?  I see this error message, Dyson exits quite early, and make does not continue.  It is quite clear, isn't it?

```
Registering POST service at /vechain-api/logs/:entity
events.js:196
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::3000
    at Server.setupListenHandle [as _listen2] (net.js:1308:16)
    at listenInCluster (net.js:1356:12)
    at Server.listen (net.js:1444:7)
    at Function.listen (/usr/local/lib/node_modules/dyson/node_modules/express/lib/application.js:618:24)
    at createServer (/usr/local/lib/node_modules/dyson/lib/dyson.js:40:9)
    at bootstrap (/usr/local/lib/node_modules/dyson/lib/dyson.js:26:15)
    at Object.<anonymous> (/usr/local/lib/node_modules/dyson/bin/dyson.js:41:3)
    at Module._compile (internal/modules/cjs/loader.js:1128:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
    at Module.load (internal/modules/cjs/loader.js:983:32)
Emitted 'error' event on Server instance at:
    at emitErrorNT (net.js:1335:8)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::',
  port: 3000
}
make[1]: *** [start-mock-dyson] Error 1
make: *** [newman-mocked] Error 2
```

**Comment by catenocrypt** [comment]
Oh, I understand the problem, @kolya182 .  To be able to check the exit code of dyson, I moved it to foreground, but this way the tester does not start.

**Comment by catenocrypt** [comment]
I have added a first test to the test suite to check for dyson availability, does not continue if not available.

```
  1. response must be valid and have a body -- HAS MOCK STARTED?
```

**Comment by kolya182** [comment]
The issue is next when testing locally
1.  Dyson runs on very common port used be other apps: `3000`
2. Test still runs even if port 3000 taken by another app

Possible solution:
1. Use less common port for testing e.g for Dyson `3347` for app `8437` to avoid running mock test and on the same port with BA default ports

**Comment by catenocrypt** [comment]
Dyson port changed to 3347 (from 3000).

**Comment by hewigovens** [comment]
can you rebase master?

**Comment by catenocrypt** [comment]
> can you rebase master?

Merged.

**Comment by catenocrypt** [comment]
CI failed, wird error.  
Complains about not finding `tests/postman/blockatlas.postman_collection.json`, though file is/was there, not changed.
 
https://dev.azure.com/TrustWallet/Trust%20BlockAtlas/_build/results?buildId=26983&view=logs&j=bd9bc89c-f7ed-5d00-2307-97661833a2b1&t=51c2314a-9229-56e3-1c98-f426bb34c6d2&l=174

File: .gitattribute [hunk 1]
```diff
@@ -1 +0,0 @@
-mock/ext-api-dyson/* linguist-vendored
```
Comment (hewigovens) [inline .gitattribute:1]: this one seems not working, atlas is still a major js project

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1 +0,0 @@
-* @EnoRage @catenocrypt @hewigovens @kolya182 @prazd
```
Comment (hewigovens) [inline .github/CODEOWNERS:1]: why delete this ?

File: .gitattribute (hunk 1 above)
Comment (catenocrypt) [reply .gitattribute:1]: Hmm, file got deleted unintentionally when I merged master branch, strange.

File: .github/CODEOWNERS (hunk 2 above)
Comment (catenocrypt) [reply .github/CODEOWNERS:1]: Hmm, file got deleted unintentionally when I merged master branch, strange.

**Comment by catenocrypt** [comment]
Good catch about the unintended master merge issues, put them back.

**Comment by catenocrypt** [comment]
PR clean now.  Let's merge to avoid further conflicts.

**Pull request merged**